contracts/ScalingPriceFeed.sol

```diff
@@ -0,0 +1,100 @@
+        decimals = decimals_;
+        description = AggregatorV3Interface(underlyingPriceFeed_).description();
+
+        uint8 chainlinkPriceFeedDecimals = AggregatorV3Interface(underlyingPriceFeed_).decimals();
+        // Note: Solidity does not allow setting immutables in if/else statements
+        shouldUpscale = chainlinkPriceFeedDecimals < decimals_ ? true : false;
+        rescaleFactor = (shouldUpscale
+            ? signed256(10 ** (decimals_ - chainlinkPriceFeedDecimals))
+            : signed256(10 ** (chainlinkPriceFeedDecimals - decimals_))
+        );
+    }
+
+    /**
+     * @notice Price for a specific round
+     * @param _roundId The round id to fetch the price for
+     * @return roundId Round id from the underlying price feed
+     * @return answer Latest price for the asset in terms of ETH
+     * @return startedAt Timestamp when the round was started; passed on from underlying price feed
+     * @return updatedAt Timestamp when the round was last updated; passed on from underlying price feed
+     * @return answeredInRound Round id in which the answer was computed; passed on from underlying price feed
+     **/
+    function getRoundData(uint80 _roundId) external view returns (
+        uint80 roundId,
+        int256 answer,
+        uint256 startedAt,
+        uint256 updatedAt,
+        uint80 answeredInRound
+    ) {
+        (uint80 roundId_, int256 price, uint256 startedAt_, uint256 updatedAt_, uint80 answeredInRound_) = AggregatorV3Interface(underlyingPriceFeed).getRoundData(_roundId);
+        return (roundId_, scalePrice(price), startedAt_, updatedAt_, answeredInRound_);
+    }
+
+    /**
+     * @notice Price for the latest round
+     * @return roundId Round id from the underlying price feed
+     * @return answer Latest price for the asset in terms of ETH
+     * @return startedAt Timestamp when the round was started; passed on from underlying price feed
+     * @return updatedAt Timestamp when the round was last updated; passed on from underlying price feed
+     * @return answeredInRound Round id in which the answer was computed; passed on from underlying price feed
+     **/
+    function latestRoundData() external view returns (
+        uint80 roundId,
+        int256 answer,
+        uint256 startedAt,
+        uint256 updatedAt,
+        uint80 answeredInRound
+    ) {
+        (uint80 roundId_, int256 price, uint256 startedAt_, uint256 updatedAt_, uint80 answeredInRound_) = AggregatorV3Interface(underlyingPriceFeed).latestRoundData();
+        return (roundId_, scalePrice(price), startedAt_, updatedAt_, answeredInRound_);
+    }
+
+    function signed256(uint256 n) internal pure returns (int256) {
+        if (n > uint256(type(int256).max)) revert InvalidInt256();
+        return int256(n);
+    }
+
+    function scalePrice(int256 price) internal view returns (int256) {
+        int256 scaledPrice;
+        if (shouldUpscale) {
+            scaledPrice = price * rescaleFactor;
```

Comment:
Are we ever going to upscale prices?

It seems like the only use case for this contract is truncating an 18 decimal price feed to 8 decimals.

Would it be simpler/more efficient to have `scalePrice` always return `price / rescaleFactor`?

Reply:
Yeah maybe simpler is better. But my goal was to make this flexible *in case* there was ever a need to scale up. There's not much extra gas costs to it, so it just depends on how much we value the extra flexibility. 

Personally, I rather have the flexibility (hardcoding price feed decimals has made the WETH deployment trickier, for example), but not a super strong opinion if we think it's not worth it.